### PR TITLE
Explicitly compute null count in concatenate APIs

### PR DESCRIPTION
**Pull request by vyasr**
## Description

The total number of nulls in the output can be computed by summing the nulls in the input columns.

Contributes to #11968

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by vyasr** [comment]
@vuule locally this change passes all tests for concatenation itself but fails some tests in the Parquet reader. Would you be able to help me look into those? I am unsure if the reader is catching some edge cases that the test is missing, or if the Parquet code has some null sanitization bug that is causing the failures. I mention sanitization because the failure specifically occurs for list and struct types.

File: cpp/src/lists/copying/concatenate.cu [hunk 1]
```diff
@@ -123,15 +124,22 @@ std::unique_ptr<column> concatenate(host_span<column_view const> columns,
...
     total_list_count, has_nulls ? mask_state::UNINITIALIZED : mask_state::UNALLOCATED);
+  cudf::size_type null_count{0};
   if (has_nulls) {
     cudf::detail::concatenate_masks(columns, static_cast<bitmask_type*>(null_mask.data()), stream);
+    null_count = std::transform_reduce(
+      columns.begin(),
+      columns.end(),
+      0,
+      [](auto x, auto y) { return x + y; },
+      [](auto col) { return col.null_count(); });
```
Comment (ttnghia) [inline cpp/src/lists/copying/concatenate.cu:135]: Avoid copying the column_view objs if not needed.
```suggestion
      [](auto const& col) { return col.null_count(); });
```

**Review by ttnghia — APPROVED** [review]

**Comment by vuule** [comment]
> @vuule locally this change passes all tests for concatenation itself but fails some tests in the Parquet reader. Would you be able to help me look into those? I am unsure if the reader is catching some edge cases that the test is missing, or if the Parquet code has some null sanitization bug that is causing the failures. I mention sanitization because the failure specifically occurs for list and struct types.

@nvdbaranec

**Comment by vuule** [comment]
Tests fail because chunked Parquet reader returns columns with incorrect null count, and the new concatenate relies on this info.
Opened https://github.com/rapidsai/cudf/pull/13111 with a potential fix.

**Comment by vyasr** [comment]
Depends on #13111

**Pull request merged**